crates/core/tedge_agent/src/file_transfer_server/http_rest.rs

```diff
@@ -1,35 +1,47 @@
 use crate::file_transfer_server::error::FileTransferError;
+use anyhow::Context;
+use axum::body::StreamBody;
+use axum::routing::get;
+use axum::routing::IntoMakeService;
+use axum::Router;
 use camino::Utf8Path;
 use camino::Utf8PathBuf;
 use hyper::server::conn::AddrIncoming;
 use hyper::Body;
 use hyper::Request;
-use hyper::Response;
 use hyper::Server;
-use routerify::Router;
-use routerify::RouterService;
+use hyper::StatusCode;
+use std::io::ErrorKind;
 use std::net::IpAddr;
 use std::net::SocketAddr;
+use std::sync::Arc;
 use tedge_actors::futures::StreamExt;
 use tedge_api::path::DataDir;
 use tedge_utils::paths::create_directories;
+use tokio::fs::File;
+use tokio::io;
+use tokio::io::AsyncBufReadExt;
 use tokio::io::AsyncWriteExt;
+use tokio::io::BufReader;
+use tokio_util::io::ReaderStream;
 
+use super::error::FileTransferRequestError as Error;
+use super::request_files::FileTransferPaths;
+
+// TODO is this not just a repeat of code in tedge_config?
 const HTTP_FILE_TRANSFER_PORT: u16 = 8000;
```

Comment:
Probably outside of the scope of these changes, but it feels like we shouldn't duplicate what `tedge_config` knows here. If someone wants the default values, they could surely achieve that with a default config?